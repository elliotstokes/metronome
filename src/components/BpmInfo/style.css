.bpm-songlist {
  list-style: none;
  padding: 0;
}

.bpm-songlist-info {
  color: rgb(190, 190, 190);
  padding: 0;
  margin: 10px 0;
  font-weight: 900;
  font-size: 14px;
}