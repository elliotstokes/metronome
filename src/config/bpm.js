import bpmData from '../resources/data/bpm.json'

export const bpmInfo = bpmData.reduce((info, current) => {
  const [title, artist, bpm] = current
  if (info[bpm]) {
    info[bpm].push({artist, title})
  } else {
    info[bpm] = [{artist, title}]
  }

  return info
}, {})

export const supportedBpms = Object.keys(bpmInfo).map(bpm => Number(bpm))

export const beatLength = bpm => 60 / bpm
