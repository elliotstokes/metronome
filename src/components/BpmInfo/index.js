import React from 'react'
import './style.css'

const BpmInfo = ({ songs }) => (
  <div>
    <h2>Songs that use this BPM:</h2>
    <ul className={'bpm-songlist'}>
      { songs.map(({ title, artist }) => (
        <li key={`${title}${artist}`} className={'bpm-songlist-info'}>{`${title} (${artist})`}</li>
      ))}
    </ul>
  </div>
)

export default BpmInfo