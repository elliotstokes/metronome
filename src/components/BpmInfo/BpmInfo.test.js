import React from 'react'
import { create } from 'react-test-renderer'
import BpmInfo from '.'
import { bpmInfo, supportedBpms } from '../../config/bpm'

test('renders correctly when empty', () => {
  const component = create(<BpmInfo songs={[]} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders correctly with songs', () => {
  const component = create(<BpmInfo songs={bpmInfo[supportedBpms[0]]} />)
  expect(component.toJSON()).toMatchSnapshot()
})