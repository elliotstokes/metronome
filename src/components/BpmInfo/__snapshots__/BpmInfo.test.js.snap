// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `
<div>
  <h2>
    Songs that use this BPM:
  </h2>
  <ul
    className="bpm-songlist"
  />
</div>
`;

exports[`renders correctly with songs 1`] = `
<div>
  <h2>
    Songs that use this BPM:
  </h2>
  <ul
    className="bpm-songlist"
  >
    <li
      className="bpm-songlist-info"
    >
      Bohemian Rhapsody (Queen)
    </li>
    <li
      className="bpm-songlist-info"
    >
      Rap God (Eminem)
    </li>
  </ul>
</div>
`;
